static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx

```diff
@@ -1187,12 +957,10 @@ export class TraceTree extends TraceTreeEventDispatcher {
     return autogroupCount;
   }
 
-  static RemoveDirectChildrenAutogroupNodes(
-    root: TraceTreeNode<TraceTree.NodeValue>
-  ): number {
+  static RemoveDirectChildrenAutogroupNodes(root: BaseNode): number {
     let removeCount = 0;
 
-    TraceTree.ForEachChild(root, node => {
+    root.forEachChild(node => {
       if (isParentAutogroupedNode(node)) {
```

Comment:
Instead of checking for the node type here using `isParentAutogroupedNode` what we could do is:
- Store a set `parentAutogroupedNodes` nodes at the tree level
- Populate it when we do autogrouping 
- Traversing `tree.parentAutogroupedNodes` to remove them from the tree in `RemoveDirectChildrenAutogroupNodes`

However, it does become cumbersome as we try to remove more node types from the tree. With this pattern,
for every new node type we are willing to later remove, we will have to add a new set at the trace level.

I think using guards is the more maintainable pattern here. We look for the nodes we want to remove on the go. 

Open to suggestions.